modules/xmpp/RtxModifier.spec.js

```diff
@@ -0,0 +1,263 @@
+  return getPrimaryVideoSsrcs(parsedSdp)[0];
+};
+
+// Only handles parsing 2 scenarios right now:
+// 1) Single video ssrc
+// 2) Multiple video ssrcs in a single simulcast group
+var getPrimaryVideoSsrcs = function (parsedSdp) {
+  let videoMLine = parsedSdp.media.find(m => m.type === "video");
+  if (numVideoSsrcs(parsedSdp) === 1) {
+    return [videoMLine.ssrcs[0].id];
+  } else {
+    let simGroups = getVideoGroups(parsedSdp, "SIM");
+    if (simGroups.length > 1) {
+      return;
+    }
+    let simGroup = simGroups[0];
+    return simGroup.ssrcs.split(" ")
+      .map(ssrcStr => parseInt(ssrcStr));
+  }
+};
+
+var getVideoGroups = function (parsedSdp, groupSemantics) {
+  let videoMLine = parsedSdp.media.find(m => m.type === "video");
+  videoMLine.ssrcGroups = videoMLine.ssrcGroups || [];
+  return videoMLine.ssrcGroups
+    .filter(g => g.semantics === groupSemantics);
+};
+
+describe ("RtxModifier", function() {
+    beforeEach(function() {
+      this.rtxModifier = new RtxModifier();
+      this.transform = transform;
+      this.SDPUtil = SDPUtil;
+    });
+
+    describe ("modifyRtxSsrcs", function() {
+      describe ("when given an sdp with a single video ssrc", function() {
+        beforeEach(function() {
+          this.singleVideoSdp = SampleSdpStrings.plainVideoSdp;
+          this.primaryVideoSsrc = getPrimaryVideoSsrc(this.singleVideoSdp);
+        });
+        it ("should add a single rtx ssrc", function() {
+          // Call rtxModifier.modifyRtxSsrcs with an sdp that contains a single video
+          //  ssrc.  The returned sdp should have an rtx ssrc and an fid group.
+          let newSdpStr = this.rtxModifier.modifyRtxSsrcs(this.transform.write(this.singleVideoSdp));
+          let newSdp = transform.parse(newSdpStr);
+          let newPrimaryVideoSsrc = getPrimaryVideoSsrc(newSdp);
+          expect(newPrimaryVideoSsrc).toEqual(this.primaryVideoSsrc);
+          // Should now have an rtx ssrc as well
+          expect(numVideoSsrcs(newSdp)).toEqual(2);
+          // Should now have an FID group
+          let fidGroups = getVideoGroups(newSdp, "FID");
+          expect(fidGroups.length).toEqual(1);
+
+          let fidGroup = fidGroups[0];
+          let fidGroupPrimarySsrc = parseInt(fidGroup.ssrcs.split(" ")[0]);
+          expect(fidGroupPrimarySsrc).toEqual(this.primaryVideoSsrc);
+        });
+
+        it ("should re-use the same rtx ssrc for a primary ssrc it's seen before", function() {
+          // Have rtxModifier generate an rtx ssrc via modifyRtxSsrcs.  Then call it again
+          //  with the same primary ssrc in the sdp (but no rtx ssrc).  It should use
+          //  the same rtx ssrc as before.
+          let newSdpStr = this.rtxModifier.modifyRtxSsrcs(this.transform.write(this.singleVideoSdp));
+          let newSdp = transform.parse(newSdpStr);
+
+          let fidGroup = getVideoGroups(newSdp, "FID")[0];
+          let fidGroupRtxSsrc = parseInt(fidGroup.ssrcs.split(" ")[1]);
```

Comment:
"SDPUtil parse groups" should be used here as well ? (please fix also other occurrences in this file)